File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    for i in range(len(photo)):
        point = 0
        for j in range(len(photo[i])):
            for k in range(len(name)):
                if photo[i][j] == name[k]:
                    point += yearning[k]
        answer.append(point)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        for (int i = 0; i < photo.length; i++) {
            int point = 0;
            for (int j = 0; j < photo[i].length; j++) {
                String person = photo[i][j];
                for (int k = 0; k < name.length; k++) {
                    if (name[k].equals(person)) {
                        point += yearning[k];
                    }
                }
            }
            answer[i] = point;
        }
        
        return answer;
    }
}